packages/jest-core/src/__tests__/watch.test.js

```diff
@@ -576,6 +582,64 @@ describe('Watch mode flows', () => {
     });
   });
 
+  describe('check clear cache modules', () => {
+    const fileTargetPath = `${__dirname}/__fixtures__/hey.test.js`;
+    let hasteMapInstance;
+    let Resolver;
+    let originalClearCache;
+
+    beforeEach(() => {
+      Resolver = require('jest-resolve');
+      originalClearCache = Resolver.clearCache;
+    });
+
+    afterEach(() => {
+      Resolver.clearCache = originalClearCache;
+      hasteMapInstance.end();
+      try {
+        fs.unlinkSync(fileTargetPath);
+      } catch (e) {}
+    });
+
+    it('should correct require new files without legacy cache', async () => {
```

Comment:
I'm wondering if we can test this without mocking `clearCache`, but instead by actually executing a sequence of runs that would fail if the cache clear is commented out.
I imagine if we write a file `a.js`, do the first run so it sets `FILE` as the type, then replace the file with a directory containing an index file `a.js/index.js` and do another run, it might break without a cache clear because it's still FILE while it should now be `DIRECTORY`?

Reply:
I wanted to do it that way. But so far I have not succeeded. I set progress in this pr. I will try to do it one more time.

Reply:
Btw. Current test works well in mac os, but failed in ci(linux).

Reply:
We run mac (and windows) on azure pipeline, and it seemed to crash with some c++ error. Might not be related